examples/git_mirror.py

```diff
@@ -9,8 +9,19 @@ def main():
 
     target_ref = f"refs/heads/{branch_name}"
 
-    gh_refs = ls_remote(src_remote_url)
-    gl_refs = ls_remote(dest_remote_url)
+    # note: provide credentials in the following situations:
+    # ls_remote: source repo requires auth to read
+    # ls_remote: destination repo requires auth to read
+    # fetch_pack: source repo requires auth to read
+    # send_pack: destination repo requires auth to write
+    # username and password args are not needed otherwise
+    src_username = "<username>"
+    src_password = "<token>"
+    dest_username = "<username>"
+    dest_password = "<token>"
+
+    gh_refs = ls_remote(src_remote_url, username=src_username, password=src_password)
+    gl_refs = ls_remote(dest_remote_url, username=dest_username, password=dest_password)
```

Comment:
```suggestion
    # Authentication credentials
    # Note: Only provide credentials when authentication is required
    # src_username = "<username>"  # Uncomment if source repo requires auth
    # src_password = "<token>"     # Uncomment if source repo requires auth
    dest_username = "<username>"   # For destination repo write access
    dest_password = "<token>"      # For destination repo write access

    # List references from source repository (without authentication in this example)
    gh_refs = ls_remote(src_remote_url)

    # List references from destination repository (with authentication)
    gl_refs = ls_remote(dest_remote_url, username=dest_username, password=dest_password)
```

What would you think about rewriting this to be something like this to show both an authenticated and unauthenticated `ls_remote()` call example. (Let's also do the same for the asyncio example.)

Reply:
sounds good!